test/datahike/test/api.cljc

```diff
@@ -104,6 +104,76 @@
     (is (thrown? clojure.lang.ExceptionInfo (d/transact conn 1)))
     (is (thrown? clojure.lang.ExceptionInfo (d/transact conn {:foo "bar"})))))
 
+(deftest test-q-docs
+  (let [cfg {:store {:backend :mem
+                     :id "q"}
+             :initial-tx [[:db/add -1 :name "Ivan"]
+                          [:db/add -1 :likes "fries"]
+                          [:db/add -1 :likes "pizza"]
+                          [:db/add -1 :friend 296]]
+             :keep-history? false
+             :schema-flexibility :read}
+        _ (d/delete-database cfg)
+        _ (d/create-database cfg)
+        conn (d/connect cfg)]
+    (is (= #{["fries"] ["candy"] ["pie"] ["pizza"]}
+           (d/q '[:find ?value :where [_ :likes ?value]]
+                #{[1 :likes "fries"]
+                  [2 :likes "candy"]
+                  [3 :likes "pie"]
+                  [4 :likes "pizza"]})))
+
+    (is (= #{["fries"] ["candy"] ["pie"] ["pizza"]}
+           (d/q {:query '[:find ?value :where [_ :likes ?value]]
+                 :args [#{[1 :likes "fries"]
+                          [2 :likes "candy"]
+                          [3 :likes "pie"]
+                          [4 :likes "pizza"]}]})))
+
+    (is (= #{["pizza"]}
+           (d/q {:query '[:find ?value :where [_ :likes ?value]]
+                 :offset 2
+                 :limit 1
+                 :args [#{[1 :likes "fries"]
+                          [2 :likes "candy"]
+                          [3 :likes "pie"]
+                          [4 :likes "pizza"]}]})))
+
+    (is (= #{["candy"] ["pizza"]}
```

Comment:
I added another test to show that the input sorting is not equal the output sorting that means the returning of candy and pizza seems to be fine since offset 2 means two items are left, right?

Reply:
I see! All right then.